Update to use jdk 17 and camunda sdk 8.6.9

upgradingdave opened the pull request:
Hi Team, I performed a benchmark exercise against Camunda helm chart `11.2.1` / Zeebe `8.6.9`. I updated this project to use `spring-boot-starter-camunda-sdk` 8.6.9 accordingly.

falko commented on src/main/resources/application.properties:
```diff
@@ -1,15 +1,26 @@
+#camunda.client.mode=simple
 
+# or selfmanaged
+camunda.client.mode=selfmanaged
+camunda.client.zeebe.base-url=http://localhost:26500
+camunda.client.auth.issuer=http://localhost:8080/auth/realms/camunda-platform/protocol/openid-connect/token
+camunda.client.zeebe.grpc-address=https://dave02.gke.c8sm.com
+camunda.client.zeebe.rest-address=https://dave02.gke.c8sm.com/rest
+camunda.client.zeebe.prefer-rest-over-grpc=false
+
+camunda.client.auth.client-id=zeebe
+camunda.client.auth.client-secret=KLn2WqAIbs
+camunda.client.identity.base-url=https://dave02.gke.c8sm.com/identity
```
Great example but it should probably be commented out. The resulting Docker image should not be pre-configured for a special environment other than maybe localhost with default port and no TLS.

falko commented on src/main/resources/application.properties:
```diff
@@ -50,9 +61,10 @@ benchmark.starterId=benchmarkStarter1
-zeebe.client.worker.max-jobs-active=2000
-zeebe.client.worker.threads=100
-zeebe.client.default-job-worker-stream-enabled=true
+camunda.client.zeebe.defaults.request-timeout=30s
+camunda.client.zeebe.execution-threads=500
+camunda.client.zeebe.defaults.max-jobs-active=3000
+camunda.client.zeebe.defaults.stream-enabled=true
```
Let's not modify the defaults unless you think that fixes something.

falko reviewed (changes requested):
@upgradingdave the PR looks quite good already and I could merge today if you have chance to do some minor tweaks.

upgradingdave commented on src/main/java/org/camunda/community/benchmarks/StartPiScheduler.java:
```diff
@@ -105,7 +105,7 @@ public void startSomeProcessInstances() {
 
     @Async
-    private void startProcessInstances(long batchSize) {
+    protected void startProcessInstances(long batchSize) {
```
<img width="792" alt="Screenshot 2025-03-27 at 10 15 23 AM" src="https://github.com/user-attachments/assets/71b94ae7-3b04-438f-9d85-7e80ef135cdb" />

upgradingdave commented:
Hi Falko, I addressed your comments. Please let me know if you find anything else. Otherwise this should be good to merge. 

I also reached out to the team and should have access to merge this myself later today.
